src/views/TradingCompetition/types.tsx

```diff
@@ -16,6 +17,33 @@ export interface UserTradingInformationProps {
   canClaimNFT?: boolean
 }
 
+export interface LeaderboardDataItem {
+  rank?: number
+  address?: string
+  username?: string
+  volume?: number
+  teamId?: number
+}
+
+interface LeaderboardData {
+  total?: number
+  volume?: number
+  data?: Array<LeaderboardDataItem>
+}
+
+export interface TeamLeaderboardProps {
+  teamId?: number
+  leaderboardData?: LeaderboardData
+}
+
+export interface TeamRanksProps {
+  team1LeaderboardInformation?: TeamLeaderboardProps
+  team2LeaderboardInformation?: TeamLeaderboardProps
+  team3LeaderboardInformation?: TeamLeaderboardProps
+  globalLeaderboardInformation?: LeaderboardData
+  isGlobalLeaderboardDataComplete?: boolean
+}
```

Comment:
I think, generally speaking, if all the props are not required then it might be better to break up the component. Just something to think about.

Reply:
Yea I think you're right. This all grew arms and legs as I developed and can probably be tightened up by being more separated. I'll also look at in a follow-up PR. Might be good to pair with the data handling modifications 👌